tools/autograd/templates/VariableType.cpp

```diff
@@ -141,36 +142,37 @@ Type & VariableType::toScalarType(ScalarType s) const {
   return *getType(baseType->toScalarType(s));
 }
 TypeID VariableType::ID() const {
-  throw std::runtime_error("VariableType::ID() not implemented");
+  return static_cast<TypeID>(id_);
 }
 
 const char * VariableType::typeString() {
   return "VariableType";
 }
 
-// Pre-condition: backend/scalar_type is a valid type in the type_registry
```

Comment:
This pre-condition didn't change right? It's still a requirement.

Reply:
register_variabe_type_for can now work on any baseType, it doesn't have to be in the type_registry so I removed the comment.  Maybe the comment should actually be moved into on one of the callers of register_variable_type_for